HeterogeneousCore/AlpakaMath/test/alpaka/testVector.dev.cc

```diff
@@ -0,0 +1,265 @@
+
+      CMS_UNROLL_LOOP
+      for (int n = 0; n < nSrc; n++) {
+        z[i + n * len] = zvec[n];
+      }
+    }
+  }
+};
+
+struct axpyKernel {
+  const data_t a;
+
+  constexpr axpyKernel() : a(c) {}
+
+  ALPAKA_FN_ACC void operator()(Acc1D const& acc, const data_t* x, const data_t* y, data_t* z) const {
+    for (auto i : uniform_elements(acc)) {
+      const Vec3d xvec(x[i], x[i + len], x[i + len * 2]);
+      const Vec3d yvec(y[i], y[i + len], y[i + len * 2]);
+
+      const Vec3d zvec = cms::alpakatools::math::axpy(a, xvec, yvec);
+
+      CMS_UNROLL_LOOP
+      for (int n = 0; n < nSrc; n++) {
+        z[i + n * len] = zvec[n];
+      }
+    }
+  }
+};
+
+struct normalizeKernel {
+  ALPAKA_FN_ACC void operator()(Acc1D const& acc, data_t* x) const {
+    for (auto i : uniform_elements(acc)) {
+      Vec3d xvec(x[i], x[i + len], x[i + len * 2]);
+
+      xvec.normalize(acc);
+
+      CMS_UNROLL_LOOP
+      for (int n = 0; n < nSrc; n++) {
+        x[i + n * len] = xvec[n];
+      }
+    }
+  }
+};
+
+struct normKernel {
+  ALPAKA_FN_ACC void operator()(Acc1D const& acc, const data_t* x, data_t* r) const {
+    for (auto i : uniform_elements(acc)) {
+      const Vec3d xvec(x[i], x[i + len], x[i + len * 2]);
+
+      r[i] = xvec.norm(acc);
+    }
+  }
+};
+
+struct partialNormKernel {
+  ALPAKA_FN_ACC void operator()(Acc1D const& acc, const data_t* x, data_t* r) const {
+    for (auto i : uniform_elements(acc)) {
+      const Vec3d xvec(x[i], x[i + len], x[i + len * 2]);
+
+      r[i] = xvec.template partial_norm<Acc1D, 2>(acc);
+    }
+  }
+};
+
+struct dotKernel {
+  ALPAKA_FN_ACC void operator()(Acc1D const& acc, const data_t* x, const data_t* y, data_t* r) const {
+    for (auto i : uniform_elements(acc)) {
+      const Vec3d xvec(x[i], x[i + len], x[i + len * 2]);
+      const Vec3d yvec(y[i], y[i + len], y[i + len * 2]);
+
+      r[i] = cms::alpakatools::math::dot(xvec, yvec);
+    }
+  }
+};
+
+int main() {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  if (devices.empty()) {
+    std::cerr << "No devices available for the " EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE) " backend, "
+      "the test will be skipped.\n";
+    exit(EXIT_FAILURE);
+  }
+
+  using data_t = double;
+
+  constexpr int N = 1 << 12;
+  constexpr int nSrc = 3;
+
+  // run the test on each device
+  for (auto const& device : devices) {
+    auto queue = Queue(device);
+
+    auto x_h = make_host_buffer<data_t[]>(queue, N * nSrc);
+    auto x_d = make_device_buffer<data_t[]>(queue, N * nSrc);
+
+    auto y_h = make_host_buffer<data_t[]>(queue, N * nSrc);
+    auto y_d = make_device_buffer<data_t[]>(queue, N * nSrc);
+
+    auto z_h = make_host_buffer<data_t[]>(queue, N * nSrc);
+    auto z_d = make_device_buffer<data_t[]>(queue, N * nSrc);
+
+    auto result_d = make_device_buffer<data_t[]>(queue, N);
+
+    auto _x = x_h.data();
+    auto _y = y_h.data();
+    auto _z = z_h.data();
```

Comment:
Avoid leading underscore (2.14 in https://cms-sw.github.io/cms_coding_rules.html#2--naming-rules-1)